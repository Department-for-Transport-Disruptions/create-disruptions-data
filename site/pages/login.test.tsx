import { render, cleanup } from '@testing-library/react';
import { afterEach, describe, expect, it } from 'vitest';
import Login, { LoginPageProps } from './login.page';

const blankInputs: LoginPageProps = {
    errors: [],
    inputs: {},
};

const withInputsAndNoErrors: LoginPageProps = {
    errors: [],
    inputs: {
        email: "dummyUser@gmail.com",
        password: "dummyPassword",
    },
};

const withInputsAndErrors: LoginPageProps = {
    ...withInputsAndNoErrors,
    errors: [{ errorMessage: "Enter a valid email address", id: "email" }],
};

afterEach(() => {
    cleanup();
});

describe('Login', () => {
    it('should render correctly when there are no inputs', () => {
        const { asFragment } = render(<Login {...blankInputs} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly with inputs and no errors', () => {
        const { asFragment } = render(<Login {...withInputsAndNoErrors} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly with inputs and errors', () => {
        const { asFragment } = render(<Login {...withInputsAndErrors} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
