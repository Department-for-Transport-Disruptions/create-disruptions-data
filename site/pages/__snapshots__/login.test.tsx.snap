// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login > should render correctly when there are no inputs 1`] = `
<DocumentFragment>
  <div
    className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
  >
    <p
      className="govuk-phase-banner__content"
    >
      <strong
        className="govuk-tag govuk-phase-banner__content__tag"
      >
        Beta
      </strong>
      <span
        className="govuk-phase-banner__text"
      >
        This is a new service. Help us improve it and
         
        <a
          className="govuk-link"
          href="/feedback"
          id="feedback-link"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          give your feedback (opens in new tab)
        </a>
        .
      </span>
    </p>
  </div>,
  <div
    class="govuk-width-container"
  >
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        class="govuk-heading-xl"
      >
        Sign in
      </h1>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/login"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <h2
              class="govuk-heading-m w-3/4"
            >
              Enter your Create Transport Disruption Data account details to sign in
            </h2>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="email"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="email-input"
                >
                  Email address
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w-3/4"
                    id="email-input"
                    maxlength="100"
                    name="email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="password"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="password-input"
                >
                  Password
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w-3/4"
                    id="password-input"
                    maxlength="100"
                    name="password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            <button
              class="govuk-button mt-8"
            >
              Sign In
            </button>
          </form>
        </div>
        <div
          class="govuk-grid-column-one-third"
        >
          <div>
            <h2
              class="govuk-heading-m"
            >
              Forgot your password?
            </h2>
            <a
              class="govuk-link govuk-!-font-size-19"
              href="/forgot-password"
            >
              Reset your password
            </a>
          </div>
        </div>
      </div>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Login > should render correctly with inputs and errors 1`] = `
<DocumentFragment>
  <div
    className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
  >
    <p
      className="govuk-phase-banner__content"
    >
      <strong
        className="govuk-tag govuk-phase-banner__content__tag"
      >
        Beta
      </strong>
      <span
        className="govuk-phase-banner__text"
      >
        This is a new service. Help us improve it and
         
        <a
          className="govuk-link"
          href="/feedback"
          id="feedback-link"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          give your feedback (opens in new tab)
        </a>
        .
      </span>
    </p>
  </div>,
  <div
    class="govuk-width-container"
  >
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        class="govuk-heading-xl"
      >
        Sign in
      </h1>
      <div
        class="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/login"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <h2
              className="govuk-heading-m w-3/4"
            >
              Enter your Create Transport Disruption Data account details to sign in
            </h2>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="email"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="email-input"
                >
                  Email address
                </label>
                <div
                  className=""
                >
                  <input
                    className="govuk-input w-3/4"
                    defaultValue="dummyUser@gmail.com"
                    disabled={false}
                    id="email-input"
                    maxLength={100}
                    name="email"
                    onChange={[Function]}
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="password"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="password-input"
                >
                  Password
                </label>
                <div
                  className=""
                >
                  <input
                    className="govuk-input w-3/4"
                    defaultValue="dummyPassword"
                    disabled={false}
                    id="password-input"
                    maxLength={100}
                    name="password"
                    onChange={[Function]}
                    type="password"
                  />
                </div>
              </div>
            </div>
            <button
              className="govuk-button mt-8"
            >
              Sign In
            </button>
          </form>
        </div>
        <div
          className="govuk-grid-column-one-third"
        >
          <div>
            <h2
              className="govuk-heading-m"
            >
              Forgot your password?
            </h2>
            <a
              className="govuk-link govuk-!-font-size-19"
              href="/forgot-password"
            >
              Reset your password
            </a>
          </div>
        </div>
      </div>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`login > should render correctly with inputs and with errors 1`] = `
[
  <div
    className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
  >
    <p
      className="govuk-phase-banner__content"
    >
      <strong
        className="govuk-tag govuk-phase-banner__content__tag"
      >
        Beta
      </strong>
      <span
        className="govuk-phase-banner__text"
      >
        This is a new service. Help us improve it and
         
        <a
          className="govuk-link"
          href="/feedback"
          id="feedback-link"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          give your feedback (opens in new tab)
        </a>
        .
      </span>
    </p>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        className="govuk-heading-xl"
      >
        Sign in
      </h1>
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/login"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <div
              aria-labelledby="error-summary-title"
              class="govuk-error-summary"
              data-module="govuk-error-summary"
              role="alert"
              tabindex="-1"
            >
              <h2
                class="govuk-error-summary__title"
                id="error-summary-title"
              >
                There is a problem
              </h2>
              <div
                class="govuk-error-summary__body"
              >
                <ul
                  class="govuk-list govuk-error-summary__list"
                >
                  <li>
                    <a
                      href="#email"
                    >
                      Enter a valid email address
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <h2
              class="govuk-heading-m w-3/4"
            >
              Enter your Create Transport Disruption Data account details to sign in
            </h2>
            <div
              class="govuk-form-group govuk-form-group--error"
            >
              <div
                class="govuk-form-group"
                id="email"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="email-input"
                >
                  Email address
                </label>
                <div
                  class=""
                >
                  <span
                    class="govuk-error-message"
                    id="email-error"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Error: 
                    </span>
                    Enter a valid email address
                  </span>
                  <input
                    aria-describedby="email-error"
                    class="govuk-input w-3/4 govuk-input--error"
                    id="email-input"
                    maxlength="100"
                    name="email"
                    type="text"
                    value="dummyUser@gmail.com"
                  />
                </div>
              </div>
            </div>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="password"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="password-input"
                >
                  Password
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w-3/4"
                    id="password-input"
                    maxlength="100"
                    name="password"
                    type="password"
                    value="dummyPassword"
                  />
                </div>
              </div>
            </div>
            <button
              class="govuk-button mt-8"
            >
              Sign In
            </button>
          </form>
        </div>
        <div
          class="govuk-grid-column-one-third"
        >
          <div>
            <h2
              class="govuk-heading-m"
            >
              Forgot your password?
            </h2>
            <a
              class="govuk-link govuk-!-font-size-19"
              href="/forgot-password"
            >
              Reset your password
            </a>
          </div>
        </div>
      </div>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Login > should render correctly with inputs and no errors 1`] = `
<DocumentFragment>
  <div
    class="app-phase-banner__wrapper"
  >
    <div
      class="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        class="govuk-phase-banner__content"
      >
        <strong
          class="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          class="govuk-phase-banner__text"
        >
          This is a new service â your 
          <a
            class="govuk-link"
            href="/feedback"
            id="feedback-link"
          >
            feedback
          </a>
           will help us to improve it.
        </span>
      </p>
    </div>
  </div>
  <div
    class="govuk-width-container"
  >
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        class="govuk-heading-xl"
      >
        Sign in
      </h1>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/login"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <h2
              class="govuk-heading-m w-3/4"
            >
              Enter your Create Transport Disruption Data account details to sign in
            </h2>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="email"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="email-input"
                >
                  Email address
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w-3/4"
                    id="email-input"
                    maxlength="100"
                    name="email"
                    type="text"
                    value="dummyUser@gmail.com"
                  />
                </div>
              </div>
            </div>
            <div
              class="govuk-form-group"
            >
              <div
                class="govuk-form-group"
                id="password"
              >
                <label
                  class="govuk-label govuk-label--s"
                  for="password-input"
                >
                  Password
                </label>
                <div
                  class=""
                >
                  <input
                    class="govuk-input w-3/4"
                    id="password-input"
                    maxlength="100"
                    name="password"
                    type="password"
                    value="dummyPassword"
                  />
                </div>
              </div>
            </div>
            <button
              class="govuk-button mt-8"
            >
              Sign In
            </button>
          </form>
        </div>
        <div
          class="govuk-grid-column-one-third"
        >
          <div>
            <h2
              class="govuk-heading-m"
            >
              Forgot your password?
            </h2>
            <a
              class="govuk-link govuk-!-font-size-19"
              href="/forgot-password"
            >
              Reset your password
            </a>
          </div>
        </div>
      </div>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
